programs/management/commands/import-profiles.py

```diff
@@ -0,0 +1,146 @@
+from django.core.management.base import BaseCommand, CommandError
+from programs.models import *
+
+import urllib2
+import ssl
+import json
+from progress.bar import ChargingBar
+from tabulate import tabulate
+
+class Command(BaseCommand):
+    help = """
+Imports URLs for ProgramProfiles from a WordPress blog
+    """
+    path = None
+    profile_type = None
+    set_primary = True
+
+    page  = 0
+    pages = 0
+
+    degrees_found = 0
+    degrees_processed = 0
+    degrees_matched = 0
+    degrees_skipped = 0
+
+    profiles_created = 0
+    profiles_updated = 0
+    profiles_skipped = 0
+
+    progress_bar = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'path',
+            type=str,
+            help='The URL of the WordPress blog to import profiles from.'
+        )
+
+        parser.add_argument(
+            'profile_type',
+            type=str,
+            help='The profile type to assign the profiles to.'
+        )
+
+        parser.add_argument(
+            '--primary',
+            type=bool,
+            dest='primary',
+            default=True
+        )
+
+    def handle(self, *args, **options):
+        self.path = options['path']
+        profile_type = options['profile_type']
+        self.set_primary = options['primary']
+
+        try:
+            self.profile_type = ProgramProfileType.objects.get(name=profile_type)
+        except:
+            raise CommandError("\"{0}\" is not a valid ProgramProfileType.".format(profile_type))
+
+        self.import_profiles()
+
+        if self.progress_bar:
+            self.progress_bar.finish()
+
+        self.print_stats()
+
+    def import_profiles(self):
+        context = ssl._create_unverified_context()
+
+        response = urllib2.urlopen(self.path, context=context)
+
+        headers = dict(response.info())
+
+        if headers.has_key('x-wp-totalpages'):
+            self.pages = int(headers['x-wp-totalpages'])
+
+        if headers.has_key('x-wp-total'):
+            self.degrees_found = int(headers['x-wp-total'])
+            self.progress_bar = ChargingBar('Processing', max=self.degrees_found)
+
+        programs = json.loads(response.read())
+        self.process_page(programs)
+
+        if self.pages > 1:
+            for page in xrange(2, self.pages + 1):
```

Comment:
Apparently `xrange()` is removed in python 3--may want to consider using `range()` instead for forward compatibility